server/datastore/s3/installer_test.go

```diff
@@ -0,0 +1,84 @@
+package s3
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInstallerExists(t *testing.T) {
+	ctx := context.Background()
+	store := setupInstallerStore(t, "installers", "random-prefix")
+	defer cleanupStore(t, store)
+
+	t.Run("returns true for existing installers", func(t *testing.T) {
+		installers := seedInstallerStore(t, store, "enroll-secret")
+
+		for _, i := range installers {
+			exists, err := store.Exists(ctx, *i)
+			require.NoError(t, err)
+			require.Equal(t, exists, true)
+		}
+	})
+
+	t.Run("returns false for non-existing installers", func(t *testing.T) {
+		i := fleet.Installer{
+			EnrollSecret: "non-existent",
+			Kind:         "pkg",
+			Desktop:      false,
+		}
+		exists, err := store.Exists(ctx, i)
+		require.Error(t, err)
+		require.Equal(t, exists, false)
+
+		i = fleet.Installer{
+			EnrollSecret: "non-existent",
+			Kind:         "pkg",
+			Desktop:      true,
+		}
+		exists, err = store.Exists(ctx, i)
+		require.Error(t, err)
+		require.Equal(t, exists, false)
+	})
+}
+
+func TestGetInstaller(t *testing.T) {
+	ctx := context.Background()
+	store := setupInstallerStore(t, "installers", "random-prefix")
+	defer cleanupStore(t, store)
```

Comment:
nit: You can run `t.Cleanup(func() { cleanupStore(t, store) })` in `setupInstallerStore` itself (that way the caller doesn't need to worry about cleanups).

Reply:
really good idea, 👍